app/routes/verify-discord.ts

```diff
@@ -0,0 +1,65 @@
+import { ActionFunction, LoaderFunction, redirect } from "@remix-run/node";
+import DiscordOauth from "discord-oauth2";
+import { destroySession, getCurrentUser, getSession } from "~/session.server";
+import { getDiscordAccessToken, getDiscordUserId } from "~/discord.server";
+import { getProfileRouteForUser } from "~/utils/routes";
+import { updateUser } from "~/database.server";
+
+export const loader: LoaderFunction = async ({ request }) => {
+  const session = await getSession(request.headers.get("Cookie"));
```

Comment:
**Thought/suggestion (non-blocking)**: The code to get the session and currentUser, and redirect to `/login` if the user is not logged in feels like it will repeat a lot. Does @remix-run have any concept of middleware? That feels like the place for this sort of thing?

Reply:
I agree, and they do! https://linear.app/codesee/issue/ENG-615/middleware-for-user-auth